asv_bench/benchmarks/indexing_engines.py

```diff
@@ -9,6 +9,11 @@
 
 from pandas._libs import index as libindex
 
+from pandas import (
+    NA,
+    Series,
```

Comment:
This file shouldn’t depend on series (see docstring)

Reply:
Thx missed that docstring

We don't need the actual array since we are passing the underlying NumPy arrays anyway, we can just construct the mask and the data separately